src/librustc/hir/map/mod.rs

```diff
@@ -1018,7 +1018,7 @@ impl Named for ImplItem<'_> {
     }
 }
 
-pub(super) fn index_hir<'tcx>(tcx: TyCtxt<'tcx>, cnum: CrateNum) -> &'tcx IndexedHir<'tcx> {
+pub(super) fn index_hir(tcx: TyCtxt<'_>, cnum: CrateNum) -> &IndexedHir<'_> {
```

Comment:
I don't like removing this lifetime annotation specifically, since `'tcx` makes it clear that the returned reference lives in a global arena.